### ISSUE TITLE
Issue due date is always one day plus

### ISSUE THREAD

**Issue by neolithos** [go-gitea/gitea#5890]
- Gitea version (or commit ref): 8006b1b
- Git version: 
- Operating system: Windows 2012 R2 (English, Timezone Germany/Berlin)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant

## Description
I set a issue due date, and it shows the date in views one day plus.
- In the database I see the correct timestamp, 1549411199 -> 2019-02-05T23:59:59.
- If I open the editor, there is also the correct stamp -> correct.
- In issue overview, history and the issue it self I see the date always one day plus -> wrong.
- Over due calculation -> correct.

## Screenshots
![issue_due](https://user-images.githubusercontent.com/9724420/51923315-fcac9900-23ea-11e9-8cc6-59ae4d551805.png)

## Code

I would fix it, but I have no glue, what is indented. So, I give the lines they are responsible.

### Save
https://github.com/go-gitea/gitea/blob/8917d66571a95f3da232a0c27bc1300210d10fde/routers/api/v1/repo/issue.go#L293-L296

https://github.com/go-gitea/gitea/blob/8917d66571a95f3da232a0c27bc1300210d10fde/routers/api/v1/repo/issue.go#L413-L419

### Get
https://github.com/go-gitea/gitea/blob/8917d66571a95f3da232a0c27bc1300210d10fde/models/issue.go#L384

`AsTimePtr` add's the local timezone, if i understand it correct.

**Comment by zeripath** [comment]
It's intended that the deadline should be the end of the day so the raw deadline datetime has 23:59:59 added to it.

It sounds like the deadline date isn't being corrected for your timezone. Of course, I'm on GMT so I can't easily replicate this... In fact it looks like your server thinks it's in UTC so it's reporting a deadline datetime that would push it into the next day for you.

**Comment by neolithos** [comment]
Then `AsTimePtr` is the wrong function to decode the unix time.
Because, this function adds the local time zone. And this is +1h, the next day.

**Comment by zeripath** [comment]
Heya, sorry was busy with other things. Feel free to put a pr up. I guess the question is how to test this what we don't want is to be in the opposite situation.

My personal feeling is that we shouldn't be using datetimes to represent dates so if I get involved in writing a pr those will have to die.

---

Sorry fat fingers with the close.

**Comment by dimpfelmoser** [comment]
As  of today I am still stumbling over this problem. Gitea Version: 1.11.4.

A simple solution would be to add 12:00 hours to the date rather than 23:59:59.

Then the timezone can shift 11:59:59 in both directions.

The time per se is not needed anyways, only the date.

```
deadline = time.Date(form.Deadline.Year(), form.Deadline.Month(), form.Deadline.Day(), 
 		12, 00, 00, 0, form.Deadline.Location()) 
```

I do not know how to create PRs.

Marc

**Comment by 6543** [comment]
@dimpfelmoser <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request> can help you :)

and for the issue 🤔 instead of calculating afterwards, we may should calculate the correct time on due-date creation and insert this one ...

**Comment by jswolf19** [comment]
Is there a compelling use case to include time (as in, is the time being used somewhere?)? I think it would be easier to consider this a date unless there is a compelling reason to include time.
If the time is necessary, then perhaps a timezone setting per repository or org. or whatever is necessary (if it doesn't already exist).

**Comment by zeripath** [comment]
Closing as fixed by #11697
